azkaban-web-server/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java

```diff
@@ -1391,9 +1404,18 @@ private void handleProjectLogsPage(final HttpServletRequest req,
         page.add("createTimestamp",project.getCreateTimestamp());
         page.add("lastModifiedTimestamp",project.getLastModifiedTimestamp());
         page.add("lastModifiedUser",project.getLastModifiedUser());
+
+        // params for project upload
+        // show if a project has prod identifier
         page.add("projectUploadLock", uploadPrivilegeUser != null && project.isUploadLocked());
         page.add("adhocUpload", project.isAdhocUploadEnabled());
         page.add("showUploadLockPanel", uploadPrivilegeUser != null);
+        // only show adhocUpload changeable button when this feature is enabled
+        page.add("showAdhocUploadFeature",
+            uploadPrivilegeUser != null && disableAdhocUploadWhenProjectUploadLocked);
+        // hide upload project button when project prod identifier is set
+        page.add("hideUploadProjectButton",
+            uploadPrivilegeUser != null && disableAdhocUploadWhenProjectUploadLocked && project.isUploadLocked());
```

Comment:
I see these lines are repeated several times in the same file - do you think we can have a common function to populate the common fields for `page`?

Reply:
these pages has different properties, but I do extract what it makes sense to project sidebars. Project sidebar is shared component to those different web pages.

Reply:
Thanks for making the changes.